relays/clients/substrate/src/client.rs

```diff
@@ -52,6 +53,14 @@ pub struct Client<C: Chain> {
 	client: RpcClient,
 	/// Genesis block hash.
 	genesis_hash: C::Hash,
+	/// If several tasks are preparing their transactions simultaneously, they may get the same
+	/// response (nonce) from the `next_account_index` call. So one of transactions will be rejected
+	/// from the pool. This lock is here to prevent situations like that.
+	///
+	/// Note that it's just lock, not the `HashMap<AccountId, Index>`. This is because querying index
+	/// does not necessarily mean that transaction with this nonce will be submitted. If we fail to
+	/// submit it, we may end up with incorrect (future) nonce in the map => infinite relay loop.
+	next_account_index_lock: Arc<Mutex<()>>,
```

Comment:
Would this happen if we have both the on-demand relay and the message relay trying to submit transactions at the same time (e.g one submitting a new header and one submitting a new message from the same account?)

Reply:
Yes - there will be a collision, so this lock is actually useless. This needs to be fixed